cli/deployment/api_client.go

```diff
@@ -11,73 +11,233 @@ import (
+	}
+
 	args := []string{
 		"deployment-manager",
 		"deployments",
-		"create",
+		action,
 		deployment.config.Name,
 		"--config",
 		deployment.configFile,
 		"--project",
-		deployment.config.Project,
+		deployment.config.GetProject(),
 	}
-	_, err := runGCloud(args...)
+
+	if preview {
+		args = append(args, "--preview")
+	}
+
+	output, err := runGCloud(args...)
 	if err != nil {
-		log.Printf("Failed to create deployment: %v, error: %v", deployment, err)
-		return nil, err
+		log.Printf("failed to %s deployment: %v, error: %v", action, deployment, err)
+		return output, err
+	}
+
+	if !preview {
+		outputs, err := GetOutputs(deployment.config.GetProject(), deployment.config.Name)
+		if err != nil {
+			log.Printf("on %s action, failed to get outputs for deployment: %v, error: %v", action, deployment, err)
+			return output, err
+		}
+		deployment.Outputs = outputs
+	}
+	return output, nil
+}
+
+/*
+cancel update/create/delete preview with gcloud deployments cancel-preview command,
+in case of cancel preview of create action, need to clean deployment and run Delete() after CancelPreview()
+*/
+func CancelPreview(deployment *Deployment) (string, error) {
+	args := []string{
+		"deployment-manager",
+		"deployments",
+		"cancel-preview",
+		deployment.config.Name,
+		"--project",
+		deployment.config.GetProject(),
+		"-q",
 	}
-	outputs, err := GetOutputs(deployment.config.Name, deployment.config.Project)
+	output, err := runGCloud(args...)
 	if err != nil {
-		log.Printf("Failed to get outputs for deployment: %v, error: %v", deployment, err)
-		return nil, err
+		log.Printf("failed to cancel preview, error: %v", err)
+		return output, err
 	}
-	deployment.Outputs = outputs
-	return deployment, nil
+	return output, nil
 }
 
-func parseOutputs(data string) (map[string]string, error) {
+// ApplyPreview function apply changes made before with --preview flag
+func ApplyPreview(deployment *Deployment) (string, error) {
+	args := []string{
+		"deployment-manager",
+		"deployments",
+		"update",
+		deployment.config.Name,
+		"--project",
+		deployment.config.GetProject(),
+		"-q",
+	}
+	output, err := runGCloud(args...)
+	if err != nil {
+		log.Printf("failed to apply preview, error: %v", err)
+		return output, err
+	}
+	return output, nil
+}
+
+// Delete function removed Deployment passed into it as parameter.
+// Boolean preview param define if changes have to be previewed before apply.
+func Delete(deployment *Deployment, preview bool) (string, error) {
+	args := []string{
+		"deployment-manager",
+		"deployments",
+		"delete",
+		deployment.config.Name,
+		"--project",
+		deployment.config.GetProject(),
+		"-q",
+	}
+	if preview {
+		args = append(args, "--preview")
+	}
+	output, err := runGCloud(args...)
+	if err != nil {
+		log.Printf("failed to get deployment manifest: %v", err)
+		return output, err
+	}
+	return output, nil
+}
+
+func GetStatus(deployment *Deployment) (Status, error) {
```

Comment:
Please add a doc comment for this exported function as described at https://golang.org/doc/effective_go.html#commentary